src/client/Microsoft.Identity.Client/AppConfig/IManagedIdentity.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.AppConfig
+{
+    /// <summary>
+    /// Interface for managed identity configuration supported by MSAL.
+    /// </summary>
+    public interface IManagedIdentity
```

Comment:
Empty interface here seems odd. And in general interfaces are difficult to:

1. version, because you can't add methods to a public interface
2. reason with, because app developers think they have to create an object which implements this interface. And sometimes they actually do (e.g. we had a number of bugs from ppl trying to implement `IAccount`, which is a bad interface)


I'm thinking more of a data class:

```csharp
public class ManagedIdentityDescription
    {
        private bool _isSystemAssigned = false;
        
        private ManagedIdentityDescription(bool isSystemAssigned)
        {
            _isSystemAssigned = isSystemAssigned;
        }

        public static ManagedIdentityDescription SystemAssigned { get; } 
            = new ManagedIdentityDescription(true); // this ensures that we don't create a new instance every time

        public static ManagedIdentityDescription UserAssignedFromClientId(string clientId)
        {
            return new ManagedIdentityDescription(false) { ClientId = clientId };
        }

        public static ManagedIdentityDescription UserAssignedFromResourceId(string resourceId)
        {
            return new ManagedIdentityDescription(false) { ResourceId = resourceId };
            
        }

        public string ClientId { get; private set; }
        public string ResourceId { get; private set; }

    }
```
CC @mjcheetham  as we discussed this

Reply:
Like the data class approach above. One more suggestion is that the data attributes shall better be `UserAssignedIdType` and `Id` which always have valid value, rather than a bunch of different IDs that might default to NULL. Such a "type + id" data model also matches to the proposal 6 in our ApiReview PR.

Reply:
Updated to ManagedIdentityConfiguration. Thanks!